go/mcap/cmd/doctor.go

```diff
@@ -0,0 +1,178 @@
+package cmd
+
+import (
+	"os"
+	"io"
+	"bytes"
+	"fmt"
+	"encoding/binary"
+
+	"github.com/fatih/color"
+	"github.com/foxglove/mcap/go/libmcap"
+	"github.com/spf13/cobra"
+)
+
+type mcapRecordReader struct {
+	reader       io.Reader
+}
+
+type RawRecord struct {
+	OpCode libmcap.OpCode
+	Bytes []byte
+}
+
+func (this *mcapRecordReader) Warn(format string, v ...interface{}) {
+	color.Yellow(format, v...)
+}
+
+func (this *mcapRecordReader) Error(format string, v ...interface{}) {
+	color.Red(format, v...)
+}
+
+func (this *mcapRecordReader) Fatal(v ...interface{}) {
+	color.Set(color.FgRed)
+	fmt.Println(v...)
+	color.Unset()
+	os.Exit(1)
+}
+
+func (this *mcapRecordReader) Fatalf(format string, v ...interface{}) {
+	color.Red(format, v...)
+	os.Exit(1)
+}
+
+
+func (this *mcapRecordReader) ConsumeMagic() {
+	magicBytes := make([]byte, len(libmcap.Magic))
+
+	bytesRead, err := this.reader.Read(magicBytes)
+	if err != nil {
+		this.Fatal(err)
+	}
+	if (bytesRead != len(libmcap.Magic)) {
+		this.Fatal("Initial MAGIC bytes missing")
+	}
+	if !bytes.Equal(magicBytes, libmcap.Magic) {
+		this.Fatal("Invalid MAGIC")
+	}
+}
+
+func (this *mcapRecordReader) NextRecord() (*RawRecord) {
+	opcodeAndLength := make([]byte, 9)
+
+	bytesRead, err := this.reader.Read(opcodeAndLength)
+	if err != nil {
+		this.Fatal(err)
+	}
+	if (bytesRead != 9) {
+		this.Fatal("No more bytes to read. Expecting Record Opcode byte and Record Length.")
+	}
+
+	opcode := libmcap.OpCode(opcodeAndLength[0])
+	recordLen := binary.LittleEndian.Uint64(opcodeAndLength[1:9])
+
+	recordBytes := make([]byte, recordLen)
+	bytesRead, err = this.reader.Read(recordBytes)
+	if err != nil {
+		this.Fatal(err)
+	}
+	if (uint64(bytesRead) != recordLen) {
```

Comment:
the parentheses around the inequality are unnecessary - I'll get some linting in here today and then this will be broken. I think gofmt will remove these, which should run if you install the go vscode plugin. Maybe not though.